### PR TITLE
update Docusaurus and other deps, fix broken styles

### PR DESCRIPTION
**Pull request by Simek**
# Why & how

Superseeds:
* #3022

This PR updates Docusaurus to latest beta (`beta.18`) and most of the other dependencies.

Due to changes in Docusaurus code, I had to adjust the config and add a bunch of fixes to our custom styles, which become broken after the update. I also had to duplicate the GA move in the config, otherwise the website won't start.

# Test plan

I have run the updated version of the website on `localhost` and I used the currently shipped website as a 1:1 comparison, so the style issues was easier to spot, however, there still might be small display problems which I was not able to catch. If you see any regression in preview deploy, please let me know!

### CODE REVIEW

File: website/versioned_docs/version-0.63/geolocation.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ To request access to location, you need to add the following line to your app's
 Android API >= 18 Positions will also contain a `mocked` boolean to indicate if position was created from a mock provider.
 
 <p>
-  Android API >= 23 Requires an additional step to check for, and request
+  Android API &gt;= 23 Requires an additional step to check for, and request
```
Comment (cortinico) [inline website/versioned_docs/version-0.63/geolocation.md:44]: Why is this needed?

**Comment by lunaleaps** [comment]
I think this looks good! Are there particular places we should look for breakages? I guess, also should we confirm that GA tracking is working (tbh not sure how to do that)

**Comment by Simek** [comment]
> Are there particular places we should look for breakages?

Generally every layout have at least one issue, sidebars too, only home page looks correct after upgrade, so it's hard to point on the specific thing. I usually test few pages, with focus on the origin diversity:
* homage (custom JS)
* versioned doc page (docs plugin)
* blog post list and blog post (blog plugin)
* community page (custom MD)

> I guess, also should we confirm that GA tracking is working (tbh not sure how to do that)

The fix has been deployed yesterday in separate PR, so you probably should see a data spike on the GA dashboard now, if everything is working correctly.

**Comment by lunaleaps** [comment]
Oh actually I do think I found an issue -- I thought it was a general issue but it's not reproing off prod: https://github.com/facebook/react-native-website/issues/3061

**Comment by Simek** [comment]
I have fixed the mobile menu hover issue and the deploy preview should be updated. 

Please let me know, if now it looks fine for you.

**Review by lunaleaps — APPROVED** [review]
